kedro/io/kedro_data_catalog.py

```diff
@@ -112,6 +114,50 @@ def __eq__(self, other) -> bool:  # type: ignore[no-untyped-def]
             other.config_resolver.list_patterns(),
         )
 
+    def keys(self, regex_search: str | None = None) -> List[str]:  # noqa: UP006
+        return self._filter_keys(regex_search)
+
+    def values(self, regex_search: str | None = None) -> List[AbstractDataset]:  # noqa: UP006
+        return [self._datasets[key] for key in self._filter_keys(regex_search)]
+
+    def items(
+        self, regex_search: str | None = None
+    ) -> List[tuple[str, AbstractDataset]]:  # noqa: UP006
+        return [(key, self._datasets[key]) for key in self._filter_keys(regex_search)]
+
+    def __iter__(self) -> str:
+        yield from self._datasets.keys()
+
+    def __getitem__(self, ds_name: str) -> AbstractDataset | None:
```

Comment:
I think this should still error `catalog["non_existent"]` while `catalog.get("non_existent")` returns `None`

Reply:
Yeah, agree